src/main/java/org/prebid/server/bidder/colossus/ColossusBidder.java

```diff
@@ -1,20 +1,143 @@
+
+    private static final TypeReference<ExtPrebid<?, ExtImpColossus>> COLOSSUS_EXT_TYPE_REFERENCE =
+            new TypeReference<>() {
+            };
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
 
     public ColossusBidder(String endpointUrl, JacksonMapper mapper) {
-        super(endpointUrl, RequestCreationStrategy.REQUEST_PER_IMP, ExtImpColossus.class, mapper);
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
     }
 
     @Override
-    protected Imp modifyImp(Imp imp, ExtImpColossus impExt) {
-        return imp.toBuilder()
-                .tagid(impExt.getTagId())
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest bidRequest) {
+        final List<HttpRequest<BidRequest>> requests = new ArrayList<>();
+        final List<BidderError> errors = new ArrayList<>();
+
+        for (Imp imp : bidRequest.getImp()) {
+            try {
+                final ExtImpColossus extImpColossus = parseImpExt(imp);
+                final Imp modifiedImp = modifyImp(imp, extImpColossus.getTagId());
+                requests.add(makeRequest(bidRequest, modifiedImp));
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+            }
+        }
+
+        return Result.of(requests, errors);
+    }
+
+    private ExtImpColossus parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), COLOSSUS_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage());
+        }
+    }
+
+    private static Imp modifyImp(Imp imp, String tagId) {
+        return imp.toBuilder().tagid(tagId).build();
+    }
+
+    private HttpRequest<BidRequest> makeRequest(BidRequest bidRequest, Imp imp) {
+        final BidRequest outgoingRequest = bidRequest.toBuilder().imp(Collections.singletonList(imp)).build();
+
+        return HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .headers(HttpUtil.headers())
+                .uri(endpointUrl)
+                .body(mapper.encodeToBytes(outgoingRequest))
+                .payload(outgoingRequest)
                 .build();
     }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall, BidRequest bidRequest) {
+        final BidResponse bidResponse;
+        try {
+            bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+        } catch (DecodeException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+
+        final List<BidderError> errors = new ArrayList<>();
+        final List<BidderBid> bids = extractBids(httpCall.getRequest().getPayload(), bidResponse, errors);
+
+        return Result.of(bids, errors);
```

Comment:
OpenrtbBidder does not collect errors, it just return result of first occured error.

Reply:
Pls, consider go code